cmake_modules/FindArrow.cmake

```diff
@@ -23,12 +23,16 @@
 #  ARROW_SHARED_LIB, path to libarrow's shared library
 #  ARROW_FOUND, whether arrow has been found
 
+if( NOT "$ENV{ARROW_HOME}" STREQUAL "")
+  set(ARROW_HOME "$ENV{ARROW_HOME}")
```

Comment:
This enables the usage of `cmake ... -DARROW_HOME=/... ` Not part of the ticket but I'm not feeling that it's worth another PR.